### PR TITLE
HTMLEditor - Add link dialog should contains info about link when cursor places on right border of link(T1157840)

### PR DESCRIPTION
**Pull request by alexanderPolosatov**

### CODE REVIEW

**Review by Krijovnick — DISMISSED** [review]

File: js/ui/html_editor/utils/toolbar_helper.js [hunk 1]
```diff
@@ -170,14 +170,31 @@ function getTargetTableNode(module, partName) {
...
+        // NOTE:
+        // See T1157840
+        // When a mouse pointer is placed on the link's right border, the quill.scroll.descendant method does not return information about the link.
+        // In this case, the "Add link" form should contains information about link.
+        [link, linkOffset] = module.quill.scroll.descendant(
+            LinkBlot,
+            range.index - 1
+        );
+        if(link) {
+            indexOffset = 1;
```
Comment (ksercs) [inline js/ui/html_editor/utils/toolbar_helper.js]: cannon we just make `linkOffset -= 1; `, don't create new `indexOffset` variable?

File: js/ui/html_editor/utils/toolbar_helper.js [hunk 2]
```diff
@@ -170,14 +170,31 @@ function getTargetTableNode(module, partName) {
...
+    [link, linkOffset] = module.quill.scroll.descendant(
         LinkBlot,
         range.index
     );
 
+    if(!link && range.length === 0) {
+        // NOTE:
+        // See T1157840
+        // When a mouse pointer is placed on the link's right border, the quill.scroll.descendant method does not return information about the link.
+        // In this case, the "Add link" form should contains information about link.
```
Comment (ksercs) [inline js/ui/html_editor/utils/toolbar_helper.js]:
```suggestion
        // In this case, we receive a necessary information from the previous index.
```

File: testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/toolbarIntegration.tests.js [hunk 3]
```diff
@@ -674,6 +674,28 @@ export default function() {
             assert.strictEqual(linkText, $textInput.val());
         });
 
+        test('Add link dialog should contains info about link when cursor places on right border of link(T1157840)', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/toolbarIntegration.tests.js]:
```suggestion
        test('Add link dialog should contains info about link when cursor is placed on right border of link(T1157840)', function(assert) {
```

File: testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/toolbarIntegration.tests.js [hunk 4]
```diff
@@ -674,6 +674,28 @@ export default function() {
...
+            const linkText = 'text';
+            const linkAddress = 'http://devexpress.com';
+            const $container = $('#htmlEditor');
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/toolbarIntegration.tests.js]: more specific, plz
```suggestion
            const $htmlEditor = $('#htmlEditor');
```

File: testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/toolbarIntegration.tests.js [hunk 5]
```diff
@@ -674,6 +674,28 @@ export default function() {
...
+            const instance = $container.dxHtmlEditor({
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/toolbarIntegration.tests.js]:
```suggestion
            const htmlEditor = $container.dxHtmlEditor({
```

File: js/ui/html_editor/utils/toolbar_helper.js (hunk 1 above)
Comment (alexanderPolosatov) [reply js/ui/html_editor/utils/toolbar_helper.js]: Yes, we can

**Review by Krijovnick — APPROVED** [review]

**Review by ksercs — APPROVED** [review]

**Pull request merged**